### PR TITLE
DietPi-Banner | Support for custom colour codes

### LINKED ISSUE #2651: DietPi-Globals/Banner | Add DietPi wide customisable colour codes and banner alignment
- https://github.com/MichaIng/DietPi/pull/2649#issuecomment-475671016
- https://github.com/MichaIng/DietPi/pull/2649#pullrequestreview-217840950

### PR DESCRIPTION
**Pull request by Fourdee**
### Status: Ready

- Adds user changeable colour support via: https://github.com/MichaIng/DietPi/pull/2649#issuecomment-475659997
- Improves visibility of secondary items (previously grey, now white with descriptions in bold white)

image

### CODE REVIEW

**Comment by Fourdee** [comment]
Users can then tweak colours set in ```/DietPi/dietpi/.dietpi-banner```
```
aCOLOUR[0]='\e[38;5;154m' #DietPi Green
aCOLOUR[1]='\e[1m' #Bold White | Main descriptions
aCOLOUR[2]='\e[90m' #Grey | Credits
aCOLOUR[3]='\e[91m' #Red | Update notifications
```

File: dietpi/func/dietpi-banner [hunk 1]
```diff
@@ -48,7 +48,19 @@
 	# Set defaults
 	aENABLED=(1 0 1 0 0 1 0 0 0 0 0 1)
 
-	GREEN_LINE=' \e[38;5;154m─────────────────────────────────────────────────────\e[0m'
+	aCOLOUR=(
+
+		'\e[0m' #Reset
```
Comment (MichaIng) [inline dietpi/func/dietpi-banner]: @Fourdee 
Perhaps `\e[0m` should stay hard corded to not allow accidental colour mess?

**Comment by MichaIng** [comment]
@Fourdee 
Nice. I already had the same idea with default white for the values, since the grey is very dark on some displays.

Do we want to add some GUI to adjust these colours? But perhaps this can wait for 6.23.

#### Other idea for v6.23:
To finalise DietPi colour customization we could make the colour codes we use globals. E.g. give them reasonable names like "failure", "success", "warning", "info" (G_DIETPI_NOTIFY), "accent" (DietPi green), "highlight" (bold white) and "announcement" (orange). These can be then used throughout our scripts to have a consistent colour theme. Warning and announcement could be also the same colour, not sure.
This also solves some possible issues where not all colour codes are supported by all terminal types, respectively they lead to different results. E.g. on VirtualBox terminal some colouors are quite funny (will post some picture later).

**Comment by Fourdee** [comment]
@MichaIng 

> Do we want to add some GUI to adjust these colours? But perhaps this can wait for 6.23.

Yep agree 👍 

> Perhaps \e[0m should stay hard corded to not allow accidental colour mess?

Good call, done.

File: dietpi/func/dietpi-banner [hunk 2]
```diff
@@ -168,11 +168,11 @@ $GREEN_LINE"
 
 	Print_Useful_Commands(){
 
-		echo -e "${aCOLOUR[2]} dietpi-launcher${aCOLOUR[0]}  $GREEN_SEPERATOR All the DietPi programs in one place.
- ${aCOLOUR[2]}dietpi-config${aCOLOUR[0]}    $GREEN_SEPERATOR Feature rich configuration tool for your device.
- ${aCOLOUR[2]}dietpi-software${aCOLOUR[0]}  $GREEN_SEPERATOR Select optimized software for installation.
- ${aCOLOUR[2]}htop${aCOLOUR[0]}             $GREEN_SEPERATOR Resource monitor.
- ${aCOLOUR[2]}cpu${aCOLOUR[0]}              $GREEN_SEPERATOR Shows CPU information and stats.\n"
```
Comment (MichaIng) [inline dietpi/func/dietpi-banner]: 👍 Just had the same idea.

**Comment by Fourdee** [comment]
@MichaIng 

Hmm, slight issue with header, it will never line up as could be master/beta/dev + 2 digit versions:
image

Only solution I see is to get length of ```SIZE=${#DIETPI_VERSION}```, then set additional blanks for each ```aDESCRIPTION``` entry. But this will impact performance.

**Comment by Fourdee** [comment]
@MichaIng 

Also want to add MOTD pulled from our server to v6.23 😄

File: dietpi/func/dietpi-banner [hunk 3]
```diff
@@ -31,36 +31,36 @@
...
+		'FQDN/hostname       '
+		'NIS domainname      '
+		'LAN IP              '
+		'WAN IP              '
+		'Freespace (RootFS)  '
 		'Freespace (userdata)'
-		'Weather (wttr.in)'
-		'Custom banner entry'
+		'Weather (wttr.in)   '
+		'Custom banner entry '
```
Comment (MichaIng) [inline dietpi/func/dietpi-banner]: @Fourdee 
Note that this entry is customizable. Perhaps we can fill up the returned value from the input box (below) with white spaces to match 20 characters?

Comment (Fourdee) [reply dietpi/func/dietpi-banner]: @MichaIng 

Hmmm, personally I believe revert back to no alignment for now due to :
https://github.com/MichaIng/DietPi/pull/2649#issuecomment-475673381

?

EDIT: Done : https://github.com/MichaIng/DietPi/pull/2649/commits/c2dce89e6ee27ce062d5316eb0f94cf857c95d7a but can change if needed.

**Comment by MichaIng** [comment]
@Fourdee 
> Also want to add MOTD pulled from our server to v6.23 😄

Yey was also thinking about this for a long time. We could even use this to inform about current bugs/issues found, based on device and installed software, to reduce duplicate bug reports and provide quick solutions on the fly.
__________
Ready from your point of view? Then I will merge. I go through the open support requests in the forum and in case merge last Beta to be released then tomorrow evening.

File: dietpi/func/dietpi-banner (hunk 3 above)
Comment (MichaIng) [reply dietpi/func/dietpi-banner]: @Fourdee 
Yeah I think it's okay for now. Space calculation time (e.g. via `$(( 20 - ${#var} ))` ) for this two strings (Custom banner entry as well) should be only marginal compared to the external commands we use to obtain banner values, especially those including www access (weather/WAN IP).

But lets move this fine tuning + performance testing to v6.23

**Pull request merged**